File: SparkStreamingDemo1/src/main/scala/org/atguigu/sparkstreaming/utils/PropertiesUtil.scala
package org.atguigu.sparkstreaming.utils

import java.util.ResourceBundle

/**
 * Created by VULCAN on 2020/11/3
 */
object PropertiesUtil {

    // 绑定配置文件
    // ResourceBundle专门用于读取配置文件，所以读取时，不需要增加扩展名
    val resourceFile: ResourceBundle = ResourceBundle.getBundle("db")

    def getValue( key : String ): String = {
        resourceFile.getString(key)
    }


}

File: SparkRealtimeProject1/realtime_common/src/main/java/com/atguigu/realtime/utils/PropertiesUtil.java
package com.atguigu.realtime.utils;

import java.util.Properties;
import java.util.ResourceBundle;

public class PropertiesUtil {
    //读取配置文件，将读到的信息封装为1个ResourceBundle
    // ResourceBundle.getBundle()  读取类路径 resource 下，以xx.properties命名的文件，传入xx
    private static ResourceBundle props = ResourceBundle.getBundle("config");

    // 获取指定属性的方法
    public static String getProperty(String name) {
        return props.getString(name);
    }

    public static void main(String[] args) {
        System.out.println(getProperty("es.nodes"));
    }
}
